pkg/apis/operator/v1alpha1/type.go

```diff
@@ -0,0 +1,776 @@
+/*
+Copyright 2022 The Karmada operator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +kubebuilder:path=karmadadeployments,shortName=kmd;kmds,scope=Namespaced
+// +kubebuilder:subresource:status
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:printcolumn:name="Status",type=string,JSONPath=".status.controlPlaneReady",description=""
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp",description="CreationTimestamp is a timestamp representing the server time when this object was created. It is not guaranteed to be set in happens-before order across separate operations. Clients may not set this value. It is represented in RFC3339 form and is in UTC."
+
+// KarmadaDeployment enables declarative installation of karmada.
+type KarmadaDeployment struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired behavior of the KarmadaDeployment.
+	// +optional
+	Spec KarmadaDeploymentSpec `json:"spec,omitempty"`
+
+	// Most recently observed status of the KarmadaDeployment.
+	// +optional
+	Status KarmadaDeploymentStatus `json:"status,omitempty"`
+}
+
+// KarmadaDeploymentSpec is the specification of the desired behavior of the KarmadaDeployment.
+type KarmadaDeploymentSpec struct {
+	// HostClusterEndPoint represente host cluster access credentials, if not specified,
+	// it will be installed in the cluster where the operator is located by default.
+	// There are two way to connect cluster, one is by kubeconfig file path. two is restore
+	// the credentials in the secret.
+	// +optional
+	HostClusterEndPoint *HostClusterEndPoint `json:"hostClusterEndPoint,omitempty"`
```

Comment:
I suggest renaming `HostClusterEndPoint` to `HostClusterCredentials`, `HostCredentials`, or `HostCluster`.
Because we need not just an `endpoint` but also credentials.

Reply:
Emm, I like `HostCluster` better. How do you say, guys?

Reply:
I think `HostCluster ` is ok, here maybe have more setting for host cluster.